### PR TITLE
[FEAT] Added TSMixerx model

### PR DESCRIPTION
**Pull request by elephaint**
This PR adds the [TSMixerx](http://arxiv.org/abs/2303.06053) model into neuralforecast (in the paper referred to as TSMixerExt), which provides support for exogenous inputs, and demonstrates its application in a Multivariate_with_TSMixer example notebook.

- Added TSMixerx and AutoTSMixerx models to NeuralForecast
- Included tests in core class
- Included performance evaluation in example notebook.
- Updated some codeparts from TSMixer for consistency with TSMixerx

... attempt 2, previous branch went into merge conflict hell.

### CODE REVIEW

**Comment by elephaint** [comment]
Could you check whether I've implemented the addition of the exogenous variables in the example correctly?

File: nbs/examples/Multivariate_with_TSMixer.ipynb [hunk 1]
```diff
@@ -16,11 +16,11 @@
    "source": [
```
Comment (cchallu) [inline nbs/examples/Multivariate_with_TSMixer.ipynb:16]: <pre class="ql-syntax" spellcheck="false">Line #19.                    val_check_steps=100,
</pre><p>I believe it is missing the <code>futr_exog_list</code> parameter to add the exogenous covariates.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/921/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

File: nbs/models.tsmixerx.ipynb [hunk 2]
```diff
@@ -0,0 +1,604 @@
+{
```
Comment (cchallu) [inline nbs/models.tsmixerx.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #29.        <code>num_workers_loader</code>: int=os.cpu_count(), workers to be used by <code>TimeSeriesDataLoader</code>.

</pre><p>missing new optimizer parameters.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/921/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

File: nbs/examples/Multivariate_with_TSMixer.ipynb (hunk 1 above)
Comment (elephaint) [reply nbs/examples/Multivariate_with_TSMixer.ipynb:16]: <p>Ah, you're right, stupid from me, I thought they were picked up automatically from the dataframe but that's not the case of course if they are not specified in <code>futr_exog_list.</code></p>

File: nbs/models.tsmixerx.ipynb (hunk 2 above)
Comment (elephaint) [reply nbs/models.tsmixerx.ipynb:1]: <p>Are that these only?</p><p> </p><p><code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;optimizer = None,</code></p><p><code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;optimizer_kwargs = None,</code></p><p> </p>

Comment (elephaint) [reply nbs/models.tsmixerx.ipynb:1]: Added these two to the input args in both TSMixer and TSMixerx

File: nbs/examples/Multivariate_with_TSMixer.ipynb (hunk 1 above)
Comment (elephaint) [reply nbs/examples/Multivariate_with_TSMixer.ipynb:16]: Added `futr_exog_list `to TSMixerx in the example, and reran the experiments

**Review by cchallu — APPROVED** [review]

**Pull request merged**